conanfile.py

```diff
@@ -384,12 +378,11 @@ def requirements(self):
         if self.options.build_framework or self.options.build_pubsub:
             self.requires("util-linux-libuuid/2.39")
         if ((self.options.build_framework and self.options.framework_curlinit)
-                or self.options.build_celix_etcdlib or self.options.build_deployment_admin
+                or self.options.build_celix_etcdlib
                 or self.options.build_rsa_discovery_common or self.options.build_rsa_remote_service_admin_dfi
                 or self.options.build_launcher):
             self.requires("libcurl/[>=7.64.1 <8.0.0]")
-        if self.options.build_deployment_admin:
-            self.requires("zlib/[>=1.2.8 <2.0.0]")
+            self.requires("zlib/[>=1.2.13 <2.0.0]") #Direct dep on zlib is needed to resolve a conflict with libcurl and libzip
```

Comment:
In Conan 2.0, require-override should be done as the following:

```Python
self.requires("openssl/1.1.1t", override=True)
```

Note that the above does not make openssl a requirement, just override upstream requirements.
This is explained here: https://github.com/conan-io/conan/issues/14535#issuecomment-1687716696